app/Http/Controllers/API/VoucherController.php

```diff
@@ -181,42 +201,56 @@ private function constructResponseMessage($responses)
         }
         if ($total_submitted === 1) {
             switch ($error_type) {
-                case 'success':
+                case 'success_add':
                     return [
-                        'message' => trans('api.messages.voucher_success')
+                        'message' => trans('api.messages.voucher_success_add'),
                     ];
                     break;
+                case 'success_reject':
+                    return [
+                        'message' => trans('api.messages.voucher_success_reject')
+                    ];
                 case 'own_duplicate':
                     return [
                         'warning' => trans('api.errors.voucher_own_dupe', [
-                            'code' => $responses['own_duplicate'][0]
-                        ])
+                            'code' => $responses['own_duplicate'][0],
+                        ]),
                     ];
                     break;
                 case 'other_duplicate':
                     return [
                         'warning' => trans('api.errors.voucher_other_dupe', [
-                            'code' => $responses['other_duplicate'][0]
-                        ])
+                            'code' => $responses['other_duplicate'][0],
+                        ]),
+                    ];
+                    break;
+                case 'failed_reject':
+                    return [
+                        'warning' => trans('api.errors.voucher_failed_reject', [
+                            'code' => $responses['failed_reject'][0],
+                        ]),
                     ];
                     break;
                 case 'invalid':
                 default:
                     return [
-                        'error' => trans('api.errors.voucher_invalid')
+                        'error' => trans('api.errors.voucher_invalid'),
                     ];
                     break;
             }
         } else {
             return [
                 // Todo: This message needs work - but not this round.
-                'message' => trans('api.messages.batch_voucher_submit', [
+                'message' => trans(
+                    'api.messages.batch_voucher_submit',
+                    [
                     'success_amount' => count($responses['success']),
```

Comment:
Does this need to be `success_add` and `success_reject`. I don't think the response has `success` key anymore.

Reply:
Hmm. will check that...